simplify oss and pro tables (#1094)
returntocorp/semgrep-docs: docs/supported-languages.md

This document provides information about supported languages and language maturity definitions for the following products:

> * Semgrep OSS Engine

Reviewer: Are we moving this somewhere or removing it entirely? And if we are removing it entirely, can I ask why?

Reviewer: Oh it's because it's not a full product! Resolving.

* Semgrep Supply Chain

## Semgrep Code 

> Secure your code quickly and continuously by scanning with Semgrep Code, a fast and lightweight SAST (Static Application Security Testing) product that leverages Semgrep OSS Engine and Semgrep Pro Engine. Semgrep Code helps you quickly find and fix vulnerabiliies in your code base. 

Reviewer: Maybe add 1 line explaining Semgrep OSS Engine and Semgrep Pro Engine, then add a link?

> Secure your code quickly and continuously by scanning with Semgrep Code, a fast and lightweight SAST (Static Application Security Testing) product that leverages Semgrep OSS Engine and Semgrep Pro Engine. Semgrep Code helps you quickly find and fix vulnerabiliies in your code base. 

Reviewer: ```suggestion
Secure your code quickly and continuously by scanning with Semgrep Code, a fast and lightweight SAST (Static Application Security Testing) product that leverages Semgrep OSS Engine and Semgrep Pro Engine. Semgrep Code helps you quickly find and fix vulnerabilities in your code base. 
```

Credit to Vale :)

### Language maturity

> Semgrep Code supports over 30 lanuages, and counting! 🚀 

Reviewer: ```suggestion
Semgrep Code supports over 30 languages and counting! 🚀 
```

<SupportedLanguagesTable />

### Maturity levels

#### Language maturity factors

Semgrep OSS Engine has three maturity levels: 

> * Generally Available (GA) 

Reviewer: ```suggestion
* Generally available (GA) 
```

* Beta
* Experimental 

Their differences are outlined in the following table:

| Feature  | GA | Beta | Experimental
|----------|---------------|------------------| ----- |
| Parse Rate  | 99%+ | 95%+ | 90%+ | 
| Number of rules  | 10+ | 5+ | 0+ | 

> | Semgrep syntax | Regexp, equivalence, deep expression operators, types and typing. All features supported in Beta. | Complete metavariable support, metavariable equality. All features supported in Experimental. | Syntax, ellipsis operaor, basic metavariable functionality.|

Reviewer: ```suggestion
| Semgrep syntax | Regexp, equivalence, deep expression operators, types and typing. All features supported in Beta. | Complete metavariable support, metavariable equality. All features supported in Experimental. | Syntax, ellipsis operator, basic metavariable functionality.|
```

> ### More Information

Reviewer: ```suggestion
### More information
```

Visit the cheat sheet generation script and associated semgrep-core test files to learn more about each feature:
* [Generation script](https://github.com/returntocorp/semgrep/blob/develop/scripts/generate_cheatsheet.py)
* [`semgrep-core` test files](https://github.com/returntocorp/semgrep/tree/develop/tests)

> Visit our public dashboard to see the parse rates for each language

Reviewer: ```suggestion
Visit the Semgrep public language dashboard to see the parse rates for each language
```

  </tr>
</table>

:::info Feature and product maturity levels

> * The detailed specifications given above apply only to language support. Language maturity levels differ from feature and product maturity levels.

Reviewer: ```suggestion
* The detailed specifications previously provided apply only to language support. Language maturity levels differ from feature and product maturity levels.
```
